### DISCUSSION TITLE
Excessive connections established for brokers on Nginx

### DISCUSSION THREAD

**Discussion by Su-InShin** [edenhill/librdkafka#3843] [Q&A] [answered]
Hello!

I am using librdkafka 1.6.2 for producing messages on Openresty-1.19.9.1.
In short, when the Nginx processes run, each worker process makes broker connections as many as its partition counts, and it seems very excessive.
For example, if it produces messages to the topic which has 60 partitions, 48 (num of worker process) * 60 (num of broker's partition) = around 2880 connections established. And I think this behavior is not good for the system.
So I want to know if the broker connections can be shared through worker processes, or each worker process can establish connections to not all but just some reasonable counts.

Thanks, Regards

**Comment by Su-InShin** [comment]
I am looking at `rk_conf.sparse_connections`.

**Reply by edenhill** [reply]
The producer will make a connection to each partition leader it is producing messages to, so if messages are produced to all 60 partitions and each partition has its own leader, then there'll be 60 connections.
Usually partitions are not spread 1:1 to brokers, but each broker being leader for multiple partitions.
Maybe this is an Openresty thing?

Connections can't be shared across different producer isntances.

**Reply by Su-InShin** [reply]
I agree it might be about the way of Openresty working.
And I understand https://github.com/edenhill/librdkafka/blob/master/INTRODUCTION.md#sparse-connections is already featured on v1.6.2 with your reply.
Thank you

**Comment by Su-InShin** [comment, marked as answer]
I found out it was because of the `--with-threads` option of the Nginx.
Thanks anyway
